docs/sw2md.py

```diff
@@ -0,0 +1,325 @@
+"""Generate Markdown documents from an OpenAPI swagger file."""
+
+from __future__ import print_function
+
+import contextlib
+import functools
+import os
+import os.path
+import re
+import sys
+
+import yaml
+
+
+# JSON Reference helpers
+
+class JRefable(object):
+    """An object that can be indexed with JSON Pointers, and supports $ref."""
+    def __init__(self, data, doc=None, ref=None):
+        self.data = data
+        self.doc = doc or data
+        self.ref = ref or '/'
+        self.name = None
+
+    def __repr__(self):
+        return repr(self.data)
+
+    def wrap(self, data, ref):
+        if isinstance(data, dict):
+            if '$ref' in data:
+                ref = data['$ref']
+                ret = JRefableObject(self.doc)[ref]
+                ret.name = ref.split('/')[-1]
+                return ret
+            return JRefableObject(data, self.doc, ref)
+        if isinstance(data, list):
+            return JRefableArray(data, self.doc, ref)
+        return data
+
+class JRefableObject(JRefable):
+    """Make a dictionary into a JSON Reference-capable object."""
+    def __getitem__(self, jref):
+        if jref.startswith('#/'):
+            parts = jref[2:]
+            data = self.doc
+            ref = '/'
+        else:
+            parts = jref
+            data = self.data
+            ref = self.ref
+        for part in parts.split('/'):
+            try:
+                data = data[part]
+            except KeyError:
+                raise KeyError("{!r} not in {!r} then {!r}".format(part, self.ref, jref))
+            ref = ref + part + '/'
+        return self.wrap(data, ref=ref)
+
+    def get(self, key, default=None):
+        if key in self.data:
+            return self.wrap(self.data[key], self.ref + key + '/')
+        return default
+
+    def keys(self):
+        return self.data.keys()
+
+    def items(self):
+        for k, v in self.data.items():
+            yield k, self.wrap(v, self.ref + k.replace('/', ':') + '/')
+
+    def __contains__(self, val):
+        return val in self.data
+
+class JRefableArray(JRefable):
+    """Make a list into a JSON Reference-capable array."""
+    def __getitem__(self, index):
+        try:
+            data = self.data[index]
+        except IndexError:
+            raise IndexError("{!r} not in {!r}".format(index, self.ref))
+        return self.wrap(data, self.ref + str(index) + '/')
+
+    def __iter__(self):
+        for i, elt in enumerate(self.data):
+            yield self.wrap(elt, self.ref + str(i) + '/')
+
+
+class OutputFiles(object):
+    """A context manager to manage a series of files.
+
+    Open a new file with `open`, then access the openedx
```

Comment:
> Open a new file with `open`, then access the openedx

autocomplete mishap?

Reply:
Yes, if by "autocomplete" you mean my brain and fingers :)